gitops/snapshot.go

```diff
@@ -681,23 +683,24 @@ func PrepareSnapshot(adapterClient client.Client, ctx context.Context, applicati
 		if containerImage == "" {
 			log.Info("component cannot be added to snapshot for application due to missing containerImage", "component.Name", applicationComponent.Name)
 			continue
+		} else {
+			// if the containerImage doesn't have a valid digest, the component
+			// will not be added to snapshot
+			err := ValidateImageDigest(containerImage)
+			if err != nil {
+				log.Error(err, "component cannot added to snapshot for application due to invalid digest in containerImage", "component.Name", applicationComponent.Name)
+				return nil, errors.Join(helpers.InvalidImageDigestError(component.Name, containerImage), err)
+			}
+			snapshotComponents = append(snapshotComponents, applicationapiv1alpha1.SnapshotComponent{
+				Name:           applicationComponent.Name,
+				ContainerImage: containerImage,
+				Source:         *componentSource,
+			})
 		}
-		// if the containerImage doesn't have a valid digest, the component
-		// will not be added to snapshot
-		err := ValidateImageDigest(containerImage)
-		if err != nil {
-			log.Error(err, "component cannot added to snapshot for application due to invalid digest in containerImage", "component.Name", applicationComponent.Name)
-			continue
-		}
-		snapshotComponents = append(snapshotComponents, applicationapiv1alpha1.SnapshotComponent{
-			Name:           applicationComponent.Name,
-			ContainerImage: containerImage,
-			Source:         *componentSource,
-		})
 	}
 
 	if len(snapshotComponents) == 0 {
-		return nil, fmt.Errorf("failed to prepare snapshot due to missing valid digest in containerImage for all components of application")
+		return nil, helpers.InvalidImageDigestError(component.Name, "undefined image digest")
```

Comment:
I still don't think this error reflects what the issue is - I would say that we'd need to relay a message that the Snapshot cannot be prepared because there are no valid components that can be added to it.

Reply:
I agree. At first, I wanted to create another error for the undefined containerImage(thinking it for half one day), then to save one customized error, I use this one. So let me create a new error for this invalid component. Thanks.